### DISCUSSION TITLE
Speed Difference Between Multiple Protocols and Methods

### DISCUSSION THREAD

**Discussion by MEM-917** [XTLS/Xray-core#4224] [Performance]
First of all, thanks for all your hard work

In recent weeks, there has been a lot of chatter about the newly introduced protocol i.e. xHTTP. I haven't used it yet because of letting it get mature enough before implementing it. Before I implement it, the performance of it needs a baseline for comparison, so, I ran a couple of iperf3 tests between 2 servers and within a server itself to establish a baseline. 

Methodology:
- I installed iperf3 and proxychains on both servers.
- Both servers run on ubuntu22.04 OS. The US server is modified with xanmod kernel.
- The target server for both methods is a VPS located in the USA.
- I used proxychains for redirecting iperf3 through socks to xray listening on port 10808.
- The client VPS in the VPS-to-VPS method is located in Germany.
- The default values for nginx, haproxy are set.
- Each test has run twice and the average is been presented here.

![Protocol Speed Comparison](https://github.com/user-attachments/assets/4e4adf7a-dba2-488e-9d6f-f89f6a2484dc)

**Comment by iopq** [comment]
For XHTTP you should try with multiplexing on and off and check

**Reply by MEM-917** [reply]
It's a simple reality+vision and the SNI is the same as the domain/host itself.

**Comment by youngzyl** [comment]
What's the encryption cipher of SS in this case?

**Reply by MEM-917** [reply]
> What's the encryption cipher of SS in this case?

It's ss-2022 256gcm.
